### ISSUE TITLE
Enable zoom in add points mode

### ISSUE THREAD

**Issue by VolkerH** [napari/napari#2146]
While playing with adding points (see #2142) I noticed another opportunity to make the user interface more slick.
When in point add mode in a points layer, neither mousewheel zoom, nor drag to pan work. I do not really see a reason why these should be disabled, requiring an additional keyboard shortcut. 

* mousewheel zoom has no conflicting meaning in point add mode
* dragging could conflict with moving selected points, but a simple check whether a selection is active could take care of that.

**Comment by sofroniewn** [comment]
Support mousewheel zoom makes sense. I think with drag we were worried that sometimes you might intend to drag but accidentally add a point or sometimes you might intend to add a point but would accidentally drag, but agree we should experiment with this and see if it could be improved. Changing `self.interactive = True` here will turn interactivity back on, curious how it feels for you. We might need to disable weird things more selectively if problems arise. https://github.com/napari/napari/blob/9c13f06d20a795afb617fdb6ae408a89e40b23e9/napari/layers/points/points.py#L1360

**Comment by VolkerH** [comment]
Ok, I just tried setting `interactive = True` in the points layer. 
With the zooming, it feels exactly right. 
The panning is problematic as the event is not implemented correctly. A point should only be added when you click and release quickly, when you quick and hold I feel it should not enter a point and go into hold mode. I think I have seen that behaviour in other tools and it feels natural.

**Comment by sofroniewn** [comment]
> The panning is problematic as the event is not implemented correctly. A point should only be added when you click and release quickly, when you quick and hold I feel it should not enter a point and go into hold mode. I think I have seen that behaviour in other tools and it feels natural.

You should be able to change that here https://github.com/napari/napari/blob/fb1170a65e4704f0860eeb536d8bc7f0bac1fde8/napari/layers/points/_points_mouse_bindings.py#L86-L89. 

Instead it might look something like

```python
def add(layer, event):
    """Add a new point at the clicked position."""
    # on press
    dragged = False
    yield

    # on move
    while event.type == 'mouse_move':
        dragged = True
        yield

    # on release
    if not dragged:
        layer.add(layer.coordinates)
```

The point would then be added on release, and if dragging occurs would not be added. Curious how that feels?

**Comment by VolkerH** [comment]
Thanks for pointing me in the right direction. I'll spend an hour on that now and see whether I can get it working.

**Comment by VolkerH** [comment]
Your code snippet is already perfect IMO (when combined with setting `interactive=True`), so it only took a few minutes. I pushed it to the branch here https://github.com/VolkerH/napari/tree/points_layer_add_drag_zoom  if you want to give it a try. It still requires setting interactive in the ipython console, I didn't change the default to True yet.

**Comment by sofroniewn** [comment]
Ok nice, i can give it a try in a bit. I'd say might as well make a PR now with the new behavior. I wonder if there's any more should be thinking about before merge - do we still need all three modes "pan/zoom", "add", "select" - i think so, but just want to ask. Also are there improvements we should make to adding things in the shapes layer too? I think there it is more tricky as you can click and drag to make a new rectangle, but I just want to ask too.

**Comment by VolkerH** [comment]
Ok, I'll make a pull request.

I also changed the default behaviour to interactive and removed a test that checks that interactive is False when add mode is enabled.

I also tried to create a test that drags and checks no point is created, but that is currently not working correctly.  I don't have too much experience with testing these events. 

I'll push it now and make a PR and then I'll call it a day. If anyone wants to continue feel free to push to my branch, I'll let this rest until Monday.

**Comment by jni** [comment]
Yasssss, thanks @VolkerH for raising this, it's been bugging me all week while playing with affinder, but I was trying to clean that up before piling on the napari issues. 😂 I'm so happy getting it done was so quick — thanks @sofroniewn for the quick pointers! 🎉

**Comment by sofroniewn** [comment]
Closed by #2148. See #2159 for a followup about allowing zoom but not pan while in selection mode

**Comment by VolkerH** [comment]
I will close this now (still shows up as open for me). Thanks !
